### ISSUE TITLE
ClassificationModel.eval_model() returns Unhashable Type Error

### ISSUE THREAD

**Issue by seyonechithrananda** [ThilinaRajapakse/simpletransformers#617]
Hi @ThilinaRajapakse,

Thanks for the continued development of the excellent simple-transformers library! I had an error I wanted to share after being unable to troubleshoot how to deal with it.

**Describe the bug**
After creating an instance of a RoBERTa-based Classification Model (with pre-trained model weights from a MLM model), I trained the model over 3 epochs on a dataframe with a train_test split of 0.8. The model completes the training successfully, but returns a type error on the test_dataset dataframe object being evaluated.

**To Reproduce**
Steps to reproduce the behavior: I've attached a runnable Colab notebook to reproduce the error:
[Colab Notebook](https://colab.research.google.com/drive/1P_qDv7nvfcTtl-q4CDeoyV1vgOErzs8K?usp=sharing
)

**Expected behavior**
The model should evaluate over the test_dataset dataframe object and return an accuracy, balanced accuracy and ROC-AUC score.

**Screenshots**

<img width="1198" alt="Screen Shot 2020-08-02 at 9 44 00 PM" src="https://user-images.githubusercontent.com/46096704/89138340-48bd2a00-d509-11ea-921c-3f4558501aaf.png">

<img width="1193" alt="Screen Shot 2020-08-02 at 9 44 18 PM" src="https://user-images.githubusercontent.com/46096704/89138352-54105580-d509-11ea-9eb2-94f4f0518685.png">

**Desktop (please complete the following information):**
 - Google Colab instance with T4 GPU

**Additional context**
N/A

Thanks,
Seyone
